renderdoc/driver/gl/gl_initstate.cpp

```diff
@@ -1783,6 +1801,58 @@ template bool GLResourceManager::Serialise_InitialState<>(WriteSerialiser &ser,
                                                           GLResourceRecord *record,
                                                           const GLInitialContents *initial);
 
+bool GLResourceManager::Serialise_InitialState(WriteSerialiser &ser, ResourceId id,
+                                               GLResourceRecord *record,
+                                               const GLInitialContents *initial)
+{
+  class AutoContextShareGroupScope
```

Comment:
Creating a class to get the constructor/destructor behaviour only for this one function is unnecessary. You can replace this with function variables, in-line the `ActivateBackdoor` code where it is called and in-line the destructor after calling `Serialise_InitialState`. I think `m_ContextShareGroup` is redundant since it seems to only be used as a flag of whether we activated the context? If so you it should be changed to a bool for that to be clearer.